pkg/mq/msgstream/mq_msgstream.go

```diff
@@ -548,6 +550,30 @@ func isDMLMsg(msg TsMsg) bool {
 	return msg.Type() == commonpb.MsgType_Insert || msg.Type() == commonpb.MsgType_Delete
 }
 
+func (ms *MqTtMsgStream) continueBuffering(endTs uint64, size uint64) bool {
+	if ms.ctx.Err() != nil {
+		return false
+	}
+
+	// first run
+	if endTs == 0 {
+		return true
+	}
+
+	// pursuit mode not enabled
+	if !paramtable.Get().ServiceParam.MQCfg.EnablePursuitMode.GetAsBool() {
+		return false
+	}
+
+	// buffer full
+	if size > paramtable.Get().ServiceParam.MQCfg.PursuitBufferSize.GetAsUint64() {
+		return false
+	}
+
+	endTime, _ := tsoutil.ParseTS(endTs)
+	return time.Since(endTime) > paramtable.Get().ServiceParam.MQCfg.PursuitLag.GetAsDuration(time.Second)
```

Comment:
If the time of the current node drifts by 10 seconds, then tSafe will always add 10 seconds lag.
Is it acceptable if there is no stable clock guarantee in a distributed scenario?

Reply:
in distributed scenario, clock sync shall be a infrastructure guarantee.
plus, pursuitLag shall be configurable, maybe we could change this default value large to make it safer.